File: Android/pool/src/main/java/com/felipearpa/tyche/pool/poolscore/spotlight/PoolSpotlightItem.kt
package com.felipearpa.tyche.pool.poolscore.spotlight

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.felipearpa.tyche.pool.PoolGamblerScoreModel
import com.felipearpa.tyche.pool.poolGamblerScoreDummyModel

@Composable
fun PoolSpotlightItem(
    modifier: Modifier = Modifier,
    shimmerModifier: Modifier = Modifier,
    poolGamblerScore: PoolGamblerScoreModel
) {
    Column(modifier = modifier) {
        Text(
            text = poolGamblerScore.poolName,
            style = MaterialTheme.typography.titleLarge,
            modifier = shimmerModifier
        )
        Text(
            text = poolGamblerScore.gamblerUsername,
            style = MaterialTheme.typography.titleSmall,
            modifier = shimmerModifier
        )
    }
}

@Preview(showBackground = true)
@Composable
private fun PoolSpotlightItemPreview() {
    Surface {
        PoolSpotlightItem(
            poolGamblerScore = poolGamblerScoreDummyModel(),
            modifier = Modifier.fillMaxWidth()
        )
    }
}

File: iOS/Pool/Sources/Pool/PoolScore/PoolSpotlightItem.swift
import SwiftUI

public struct PoolSpotlightItem: View {
    let poolGamblerScore: PoolGamblerScoreModel

    public init(poolGamblerScore: PoolGamblerScoreModel) {
        self.poolGamblerScore = poolGamblerScore
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(poolGamblerScore.poolName)
                .font(.title)

            Text(poolGamblerScore.gamblerUsername)
                .font(.title3)
        }
    }
}

#Preview {
    PoolSpotlightItem(poolGamblerScore: poolGamblerScoreDummyModel())
}
